skymp5-server/cpp/server_guest_lib/ActionListener.cpp

```diff
@@ -456,8 +456,37 @@ void UseCraftRecipe(MpActor* me, const espm::COBJ* recipeUsed,
   spdlog::info("Using craft recipe with EDID {} from espm file with index {}",
                recipeUsed->GetEditorId(cache), espmIdx);
 
+  std::vector<Condition*> conditions;
+
   for (auto& condition : recipeData.conditions) {
-    // impl race, item, perk? checks
+    if (condition.IsGetItemCount()) {
+      conditions.push_back(new ItemCountCondition(
+        condition.GetDefaultData().firstParameter, condition.comparisonValue,
+        condition.GetOperator(), condition.GetFlags()));
+    } else if (condition.IsGetIsRace()) {
+      conditions.push_back(new RaceCondition(
+        condition.GetDefaultData().firstParameter, condition.comparisonValue,
+        condition.GetOperator(), condition.GetFlags()));
+    }
+  }
+
+  bool requireAnd = false;
+  for (auto& cond : conditions) {
+    if (!cond->Evaluate(me)) {
+      if (cond->GetFlags() != espm::CTDA::Flags::OR || requireAnd) {
```

Comment:
If we want checking if the flag is set, we *have* to use & instead of != no matter how:
```
static_cast<uint32_t>(cond->GetFlags()) & static_cast<uint32_t>(espm::CTDA::Flags::OR)
```

== and & is not the same. the first one will check that ALL flags are in the same state